diff --git a/Android.bp b/Android.bp
index d7c56cc80e5aab8a5d9cd261bd82041d720a97a7..87c1a57f513bdc1e1fcc7bca980886739c4ca047 100644
--- a/Android.bp
+++ b/Android.bp
@@ -294,50 +294,50 @@
 //             relative_install_path: "system/lib",
 //         },
 //         lib64: {
 //             relative_install_path: "system/lib64",
 //         },
 //     },
 //     shared_libs: [
 //         "libbase",
 //         "liblog",
 //         "librecovery_ui",
 //     ],
 //     // TARGET_RECOVERY_UI_LIB should be one of librecovery_ui_{default,wear,vr,ethernet} or a
 //     // device-specific module that defines make_device() and the exact RecoveryUI class for the
 //     // target. It defaults to librecovery_ui_default, which uses ScreenRecoveryUI.
 //     whole_static_libs: select(soong_config_variable("recovery", "target_recovery_ui_lib"), {
 //         any @ libs: libs,
 //         default: ["librecovery_ui_default"],
 //     }),
 // }
 
-// prebuilt_res {
-//     name: "recovery-resources-common-mdpi",
-//     recovery: true,
-//     install_in_root: true,
-//     relative_install_path: "images",
-//     srcs: [
-//         "res-mdpi/images/*.png",
-//     ],
-//     no_full_install: true,
-// }
+prebuilt_res {
+    name: "recovery-resources-common-mdpi",
+    recovery: true,
+    install_in_root: true,
+    relative_install_path: "images",
+    srcs: [
+        "res-mdpi/images/*.png",
+    ],
+    no_full_install: true,
+}
 
 // prebuilt_res {
 //     name: "recovery-resources-common-hdpi",
 //     recovery: true,
 //     install_in_root: true,
 //     relative_install_path: "images",
 //     srcs: [
 //         "res-hdpi/images/*.png",
 //     ],
 //     no_full_install: true,
 // }
 
 // prebuilt_res {
 //     name: "recovery-resources-common-xhdpi",
 //     recovery: true,
 //     install_in_root: true,
 //     relative_install_path: "images",
 //     srcs: [
 //         "res-xhdpi/images/*.png",
 //     ],
