diff --git a/core/envsetup.mk b/core/envsetup.mk
index 7a26485..133cf1b 100644
--- a/core/envsetup.mk
+++ b/core/envsetup.mk
@@ -176,6 +176,10 @@ ifneq (,$(findstring x86_64,$(UNAME)))
   HOST_ARCH := x86_64
   HOST_2ND_ARCH := x86
   HOST_IS_64_BIT := true
+else ifneq (,$(findstring aarch64,$(UNAME)))
+  HOST_ARCH := arm64
+  HOST_2ND_ARCH :=
+  HOST_IS_64_BIT := true
 else
 ifneq (,$(findstring i686,$(UNAME))$(findstring x86,$(UNAME)))
 $(error Building on a 32-bit x86 host is not supported: $(UNAME)!)
@@ -245,7 +249,11 @@ endif
 endif
 
 # We don't want to move all the prebuilt host tools to a $(HOST_OS)-x86_64 dir.
+ifeq ($(HOST_ARCH),arm64)
+HOST_PREBUILT_ARCH := arm64
+else
 HOST_PREBUILT_ARCH := x86
+endif
 # This is the standard way to name a directory containing prebuilt host
 # objects. E.g., prebuilt/$(HOST_PREBUILT_TAG)/cc
 # This must match the logic in get_host_prebuilt_prefix in envsetup.sh
